### ISSUE TITLE
OS X 10.11 Warning

### ISSUE THREAD

**Issue by alexey-lysiuk** [kcat/openal-soft#20]
Call to ```alcOpenDevice()``` outputs the following warning in the console while running on OS X 10.11 El Capitan:

```WARNING:  140: This application, or a library it uses, is using the deprecated Carbon Component Manager for hosting Audio Units. Support for this will be removed in a future release. Also, this makes the host incompatible with version 3 audio units. Please transition to the API's in AudioComponent.h.```

Tested on db0f29f6d8. Sound output seems to be the same as on previous versions of OS X.

Here is the complete output of **openal-info**:
```
Available playback devices:
    CoreAudio Default
Available capture devices:
    CoreAudio Default
Default playback device: CoreAudio Default
Default capture device: CoreAudio Default
ALC version: 1.1
2015-09-18 11:57:17.803 openal-info[11031:44919] 11:57:17.802 WARNING:  140: This application, or a library it uses, is using the deprecated Carbon Component Manager for hosting Audio Units. Support for this will be removed in a future release. Also, this makes the host incompatible with version 3 audio units. Please transition to the API's in AudioComponent.h.

** Info for device "CoreAudio Default" **
ALC version: 1.1
ALC extensions:
    ALC_ENUMERATE_ALL_EXT, ALC_ENUMERATION_EXT, ALC_EXT_CAPTURE,
    ALC_EXT_DEDICATED, ALC_EXT_disconnect, ALC_EXT_EFX,
    ALC_EXT_thread_local_context, ALC_SOFTX_device_clock, ALC_SOFTX_HRTF,
    ALC_SOFT_loopback, ALC_SOFTX_midi_interface, ALC_SOFT_pause_device
OpenAL vendor string: OpenAL Community
OpenAL renderer string: OpenAL Soft
OpenAL version string: 1.1 ALSOFT 1.16.0
OpenAL extensions:
    AL_EXT_ALAW, AL_EXT_BFORMAT, AL_EXT_DOUBLE, AL_EXT_EXPONENT_DISTANCE,
    AL_EXT_FLOAT32, AL_EXT_IMA4, AL_EXT_LINEAR_DISTANCE, AL_EXT_MCFORMATS,
    AL_EXT_MULAW, AL_EXT_MULAW_BFORMAT, AL_EXT_MULAW_MCFORMATS, AL_EXT_OFFSET,
    AL_EXT_source_distance_model, AL_LOKI_quadriphonic,
    AL_SOFT_block_alignment, AL_SOFT_buffer_samples, AL_SOFT_buffer_sub_data,
    AL_SOFT_deferred_updates, AL_SOFT_direct_channels, AL_SOFT_loop_points,
    AL_SOFT_MSADPCM, AL_SOFT_source_latency, AL_SOFT_source_length
EFX version: 1.0
Max auxiliary sends: 4
Supported filters:
    Low-pass, High-pass, Band-pass
Supported effects:
    EAX Reverb, Reverb, Chorus, Distortion, Echo, Flanger, Ring Modulator,
    Compressor, Equalizer, Dedicated Dialog, Dedicated LFE
Program ended with exit code: 0
```

**Comment by kcat** [comment]
It seems the functions ```FindNextComponent```, ```OpenAComponent```, ```CloseComponent```, and ```AudioHardwareGetProperty``` have been deprecated. I'm having trouble finding info on what to replace them with, though. Is there some place I can look for that?

**Comment by alexey-lysiuk** [comment]
When I'm encountering some deprecated API issue, I am searching by the function name on https://developer.apple.com site.
For instance, [this tech note](https://developer.apple.com/library/mac/technotes/tn2223/_index.html) was found while searching for ```AudioHardwareGetProperty```.
Honestly, Apple's dev site isn't comprehensive in many aspects, but it may hint with replacement candidates for obsolete functions.

**Comment by slime73** [comment]
SDL's CoreAudio backend for SDL_audio has the same problem. However it already has alternate non-deprecated codepaths in place that it uses for iOS right now, which should also work for Mac OS 10.6+.

It might be worth checking that out to see what it does: https://hg.libsdl.org/SDL/file/adb374e144d6/src/audio/coreaudio/SDL_coreaudio.c

Someone also made a patch for that SDL code to only use those deprecated codepaths when Mac OS 10.5 support is required: https://bugzilla.libsdl.org/attachment.cgi?id=2273&action=diff

**Comment by kcat** [comment]
Should now be fixed with commit 21c84bcd96a2e457b446e53d3287878e32fc77e0.
